Analytics/Classes/Internal/SEGFileStorage.m

```diff
@@ -129,20 +135,80 @@ - (NSURL *)urlForKey:(NSString *)key
 
 #pragma mark - Helpers
 
-- (id _Nullable)plistForKey:(NSString *)key
+- (id _Nullable)jsonForKey:(NSString *)key
 {
+    id result = nil;
+    
     NSData *data = [self dataForKey:key];
-    return data ? [self plistFromData:data] : nil;
+    if (data) {
+        BOOL needsConversion = NO;
+        result = [self jsonFromData:data needsConversion:&needsConversion];
+        if (needsConversion) {
```

Comment:
isn't needsConversion always going to be false here?

Reply:
No, the jsonFromData:needsConversion: method will set it's value if necessary.